### PR TITLE
log extruder and file position on pause

### PR DESCRIPTION
**Pull request by danielkucera**
* [x] Your changes are not possible to do through a plugin and relevant
    to a large audience (ideally all users of OctoPrint)
  * [x] If your changes are large or otherwise disruptive: You have
    made sure your changes don't interfere with current development by
    talking it through with the maintainers, e.g. through a
    Brainstorming ticket
  * [x] Your PR targets OctoPrint's `devel` branch if it's a completely
    new feature, or `maintenance` if it's a bug fix or improvement of
    existing functionality for the current stable version (no PRs
    against `master` or anything else please)
  * [x] Your PR was opened from a custom branch on your repository
    (no PRs from your version of `master`, `maintenance`, or `devel`
    please), e.g. `dev/my_new_feature` or `fix/my_bugfix`
  * [x] Your PR only contains relevant changes: no unrelated files,
    no dead code, ideally only one commit - rebase and squash your PR
    if necessary!
  * [x] Your changes follow the existing coding style
  * [x] If your changes include style sheets: You have modified the
    `.less` source files, not the `.css` files (those are generated
    with `lessc`)
  * [x] You have tested your changes (please state how!) - ideally you
    have added unit tests
  * [ ] You have run the existing unit tests against your changes and
    nothing broke
  * [ ] You have added yourself to the `AUTHORS.md` file :)

#### What does this PR do and why is it necessary?
It logs extruder and file position to log in case of failure during pause state.

#### How was it tested? How can it be tested by the reviewer?
I run it during printing and the log showed added values

### CODE REVIEW

File: src/octoprint/printer/standard.py [hunk 1]
```diff
@@ -1744,12 +1744,19 @@ def on_comm_print_job_paused(self, suppress_script=False, user=None):
         )
         if payload:
             eventManager().fire(Events.PRINT_PAUSED, payload)
+            if "position" in payload:
+                position = str(payload["position"])
+            else:
+                position = "unknown"
+            fileposition = str(self._comm.getFilePosition())
```
Comment (foosel) [inline src/octoprint/printer/standard.py:1751]: `self._comm` might be `None` under certain (very unlikely but still possible) circumstances, so this needs a null-check.

File: src/octoprint/printer/standard.py [hunk 2]
```diff
@@ -1744,12 +1744,19 @@ def on_comm_print_job_paused(self, suppress_script=False, user=None):
         )
         if payload:
             eventManager().fire(Events.PRINT_PAUSED, payload)
+            if "position" in payload:
+                position = str(payload["position"])
+            else:
+                position = "unknown"
```
Comment (foosel) [inline src/octoprint/printer/standard.py:1750]: We can get rid of that and instead copy the fetch of `origin` etc below

File: src/octoprint/printer/standard.py [hunk 3]
```diff
@@ -1744,12 +1744,19 @@ def on_comm_print_job_paused(self, suppress_script=False, user=None):
...
+            fileposition = str(self._comm.getFilePosition())
             self._logger_job.info(
-                "Print job paused - origin: {}, path: {}, owner: {}, user: {}".format(
+                "Print job paused - origin: {}, path: {}, owner: {}, user: {}, fileposition: {}, position: {}".format(
                     payload.get("origin"),
                     payload.get("path"),
                     payload.get("owner"),
                     payload.get("user"),
+                    fileposition,
+                    position,
```
Comment (foosel) [inline src/octoprint/printer/standard.py:1759]:
```suggestion
                    payload.get("position"),
```

**Review by foosel — CHANGES_REQUESTED** [review]
First of all, I agree that getting that into the log makes sense 👍️

However, since the log line currently pretty much copies what's inside the event (minus the position), I wonder if it wouldn't make sense to extend the event here as well to include the file position (origin and file name are already there, your current code duplicates that information as it's also part of the `getFilePosition` response). 

And then we probably should also do the same for cancelling - not that much of interested in case of an issue during pausing, but rather very much helpful for people who want to attempt to rescue a print that got cancelled.

Event extension then also needs some docs.

Also, some small issues, see individual comments.

**Comment by foosel** [comment]
Also, as I just noticed, this should be filed against `maintenance`, not `devel`. It's an improvement of existing functionality, not a biiiiiig new feature that requires a ton of testing and lengthy integration processes first.

**Comment by danielkucera** [comment]
@foosel , I've put everything into event data and included cancel event.

**Comment by danielkucera** [comment]
It's updated, please check

**Pull request merged**